src/create-text-tracks-if-necessary.js

```diff
@@ -16,14 +16,36 @@ const createTextTracksIfNecessary = function(sourceBuffer, mediaSource, segment)
 
   // create an in-band caption track if one is present in the segment
   if (segment.captions &&
-      segment.captions.length &&
-      !sourceBuffer.inbandTextTrack_) {
-    removeExistingTrack(player, 'captions', 'cc1');
-    sourceBuffer.inbandTextTrack_ = player.addRemoteTextTrack({
-      kind: 'captions',
-      label: 'cc1'
-    }, false).track;
-    player.tech_.trigger({type: 'usage', name: 'hls-608'});
+      segment.captions.length) {
+    if (!sourceBuffer.inbandTextTracks_) {
+      sourceBuffer.inbandTextTracks_ = {};
+    }
+
+    segment.captions.forEach(function(caption) {
```

Comment:
Something along those lines, just a matter of where and how to communicate that information. You're definitely right that this code exists because we haven't had a previous way of detecting which channels were in use until after we've transmuxed a segment. Since the CC tag information is optional in the manifest, having the detection and track creation here is still important. Trusting the manifest for optional information is usually a recipe for issues down the road.  I don't think the muxer would need to hold extra state as the information is already there in the captions. The transmuxer itself already needs to loop through the captions to adjust start and end times (https://github.com/videojs/mux.js/blob/master/lib/mp4/transmuxer.js#L973). This is also the last point that the transmuxer sees the captions before it sends it back to here. I think it would be reasonable to attach a bit of extra information to the `event` variable. Maybe something like `event.captionStreams` (or a more appropriate name) (Note: this would also have to be done in the `CoalesceStream` of the FLV' as well)
```
event.captionStreams = {
  CC1: false,
  CC2: false,
  CC3: false,
  CC4: false
};
```
and set the bool to true when you encounter it during that loop. Then instead of looping through the captions again here, you can just check `segment.captionStreams` and compare that to available tracks to determine if one needs to be made. I think we can avoid any breaking changes with this approach